src/ensembl/production/metadata/api/adaptors/genome.py

```diff
@@ -634,27 +636,53 @@ def fetch_genome_datasets(self,
                 if release_version:
                     genome_datasets = [gd for gd in genome_datasets if
                                        float(gd.ensembl_release.version) <= release_version]
+
+                if len(genome_datasets) > 1:
+                    logger.debug(f"{len(genome_release)} genome_datasets found")
+                    logger.debug(f"Retrieved genome_datasets {genome_release}")
+                    # this means that we have datasets that are released in both and partial,
+                    # if it's the case we pick the partial dataset because "if a dataset is provided in a partial release
+                    # for an existing genome we would prefer that dataset"
+                    # https://genomes-ebi.slack.com/archives/C010QF119N1/p1746101265211759?thread_ts=1746094298.003789&cid=C010QF119N1
+                    if not cfg.allow_unreleased:
+                        genome_datasets = [gd for gd in genome_datasets if gd.ensembl_release.release_type == "partial"]
```

Comment:
this filter is checking for release type instead of release status (unreleased partial could be returned here)?

Reply:
The logic you mention is in line `631` above, I added this condition here for tests to pass regardless of the datasets release type.